drv/psc-seq-server/src/main.rs

```diff
@@ -3,48 +3,597 @@
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
 //! Server for managing the PSC sequencing process.
+//!
+//!
+//! # General notes on PSC power supply sequencing
+//!
+//! There are rules to follow here to avoid glitching the power supplies,
+//! because glitching the power supplies here will glitch the entire rack,
+//! making you very unpopular very quickly.
+//!
+//! **`ON_L` signals to the PSUs:** we normally leave our pins high-impedance on
+//! these nets, allowing external resistors to pull them low. We only drive them
+//! to _disable_ the PSU by driving it high. To achieve this, we leave the pin
+//! configured as an input, pre-load the output value as "high," and toggle its
+//! mode register between input and output.
+//!
+//! **`PRESENT_L` signals from the PSUs:** pulled inactive-high by resistors on
+//! the board and power shelf. When these go low, assume they will bounce,
+//! because they're brought low by a physical connection between pins on our
+//! connector.
+//!
+//! **`OK` signals from the PSUs:** pulled ACTIVE-high by resistors on the
+//! board, in all caps because resistors pulling something active is somewhat
+//! unusual. The PSU drives this open drain, so it will only go low if the PSU
+//! pulls it low to indicate a problem. This implies that, if the PSU is not
+//! detected as present, you cannot trust the `OK` signal.
+//!
+//!
+//! # Intended behavior
+//!
+//! The PSC is intended to be hot swappable. If the PSC gets plugged in, this
+//! task will start anew. It will also start anew if power is applied to a
+//! previously unpowered rack.
```

Comment:
The "this task will start anew" seems to be in conflict with the sentence below "This task is not intended to restart under normal operation". I _think_ this paragraph is describing cold boot behavior vs below is runtime?

Reply:
Yeah, basically -- the PSC getting plugged in means _power is applied._ I have attempted to clarify this.